kolibri/plugins/management/assets/src/actions.js

```diff
@@ -25,19 +25,30 @@ const samePageCheckGenerator = require('kolibri.coreVue.vuex.actions').samePageC
  * the API to state in the Vuex store
  */
 
-function _userState(data) {
-  // assume just one role for now
+function _stateUser(apiUserData) {
+  // handle role representation
   let kind = UserKinds.LEARNER;
-  if (data.roles.length && data.roles[0].kind === 'admin') {
-    kind = UserKinds.ADMIN;
-  }
+
+  // makes kind the highest ranking role
+  apiUserData.roles.forEach(role => {
+    // using a switch statement. Checks all in order of heirarchy
+    switch(role.kind){
+      // sets role to admin 
+      case UserKinds.ADMIN || UserKinds.SUPERUSER:
+        kind = UserKinds.ADMIN;
+        break;
+      case UserKinds.COACH:
+        if(kind != UserKinds.ADMIN) kind = UserKinds.COACH;
+        break;
+    }
+  });
```

Comment:
Using a switch statement with conditional assignment inside a loop seems a bit sketchy: it makes assumptions about the order the roles appear in and the possible values that might exist.

I would recommend instead doing a series of if-else statements like:

```javascript
if (apiUserData.roles.includes(UserKinds.ADMIN)) {
  kind = UserKinds.ADMIN;
} else if (apiUserData.roles.includes(UserKinds.COACH)) {
  kind = UserKinds.ADMIN;
} else {
  ...
}
```

Might be a little slower but much clearer in meaning.